### DISCUSSION TITLE
Should the All-in-one and/or the RCP designer support editing *.js files?

### DISCUSSION THREAD

**Discussion by hvbtup** [eclipse-birt/birt#770] [Ideas]
With BIRT designer (be it all-in-one or the RCP designer), when you have *.js resources, these can be used in BIRT reports easily, but the BIRT IDE does not support _editing_ them. Instead, they are opened by the OS (which might even be dangerous).

I know there is support for editing *.JS in the web development tools somewhere including syntax highlighting (sorry, but I don't know more details, I'm not a frontend developer) .  I vaguely remember that I installed this manually some years ago, and I think this would be a nice addition in general.

**Comment by wimjongman** [comment]
the JS editor can be provided by the wild-web-developer project. We need to include their features.

**Comment by qs-wang** [comment]
I remember there was a build/package which include the web editor plugins years ago. Not sure if that is still there, but we can definitely include the WDT features.
